apps/web/src/blocks/HeroBlock/index.tsx

```diff
@@ -128,28 +150,52 @@ function HeroBlock({
   title,
   subTitle,
   cta,
-  input,
+  textinput,
   image,
-  blockConfig
+  blockConfig,
+  contentAlign,
+  layout
 }: HeroBlockProps) {
-  const layout = blockConfig?.layout || 'imgRight';
+  const isFullBleed =
+    typeof image === 'object' && typeof image?.imageProps?.fill === 'boolean'
+      ? image?.imageProps?.fill
+      : false;
+
+  const contentPosition = layout || 'contentLeft';
+  const alignText = contentAlign || 'left';
 
-  const className = genClassName([layout]);
   return (
-    <StyledWrapper className={className} gutter={false} fullBleed>
-      <ImageWrapper image={image} $layout={layout} />
-
-      <ContentWrapper className={className}>
-        {eyebrow && <Eyebrow>{eyebrow}</Eyebrow>}
-        {title && <Title>{title}</Title>}
-        {subTitle && <SubTitle {...subTitle} />}
-        {(input || cta) && (
-          <InputWrapper className={className}>
-            {input?.type && <Input {...input} />}
-            {cta && <CtaButton cta={cta} />}
-          </InputWrapper>
-        )}
-      </ContentWrapper>
+    <StyledWrapper
+      gutter={false}
+      {...blockConfig}
+      hidden={blockConfig?.hidden ?? false}
+    >
+      <InnerWrapper $layout={contentPosition} $isFullBleed={isFullBleed}>
+        <ContentWrapper
+          $layout={contentPosition}
+          $isFullBleed={isFullBleed}
+          $contentAlign={alignText}
+        >
+          {eyebrow && <Eyebrow>{eyebrow}</Eyebrow>}
+          {title && <Title>{title}</Title>}
+          {subTitle && <SubTitle {...subTitle} />}
+          {textinput && cta && (
+            <InputWrapper
+              onSubmit={(data) => console.log(data)}
+              cta={cta}
+              $contentAlign={alignText}
+            >
+              <TextInput
+                {...textinput}
+                name="TextInput"
```

Comment:
Hardcoding for these values because the typing for these fields needs to be updated in refract. The label and placeholder are incompatible with the typing for payload text strings.